examples/dt/uni08theta/sriov_amd-sev/edpm/openstackdataplanenodeset.yaml

```diff
@@ -0,0 +1,129 @@
+      networks:
+      - defaultRoute: true
+        fixedIP: 192.168.122.101 # CHANGEME
+        name: CtlPlane
+        subnetName: subnet1
+      - name: InternalApi
+        subnetName: subnet1
+      - name: Storage
+        subnetName: subnet1
+      - name: Tenant
+        subnetName: subnet1
+  networkAttachments:
+    - ctlplane
+  nodeTemplate:
+    ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+    managementNetwork: ctlplane
+    ansible:
+      ansibleUser: cloud-admin
+      ansiblePort: 22
+      ansibleVars:
+        # CHANGEME -- see https://access.redhat.com/solutions/253273
+        # edpm_bootstrap_command: |
+        #       subscription-manager register --username <subscription_manager_username> --password <subscription_manager_password>
+        #       podman login -u <registry_username> -p <registry_password> registry.redhat.io
+        service_net_map:
+          nova_api_network: internal_api
+          nova_libvirt_network: internal_api
+        timesync_ntp_servers:
+          - hostname: clock.redhat.com
+        # CPU pinning settings
+        edpm_kernel_args: "hugepagesz=1GB hugepages=32 default_hugepagesz=1GB iommu=pt mem_encrypt=on kvm kvm_amd.sev=1"
+        # edpm_network_config
+        # These vars are edpm_network_config role vars
+        edpm_network_config_hide_sensitive_logs: false
+        edpm_network_config_template: |
+          ---
+          {% set mtu_list = [ctlplane_mtu] %}
+          {% for network in role_networks %}
+          {{ mtu_list.append(lookup('vars', networks_lower[network] ~ '_mtu')) }}
+          {%- endfor %}
+          {% set min_viable_mtu = mtu_list | max %}
+          network_config:
+          - type: ovs_bridge
+            name: {{ neutron_physical_bridge_name }}
+            mtu: {{ min_viable_mtu }}
+            use_dhcp: false
+            dns_servers: {{ ctlplane_dns_nameservers }}
+            domain: {{ dns_search_domains }}
+            addresses:
+            - ip_netmask: {{ ctlplane_ip }}/{{ ctlplane_subnet_cidr }}
+            routes: {{ ctlplane_host_routes }}
+            members:
+            - type: interface
+              name: enp7s0
+              mtu: {{ min_viable_mtu }}
+              # force the MAC address of the bridge to this interface
+              primary: true
+          {% for network in role_networks %}
+            - type: vlan
+              mtu: {{ lookup('vars', networks_lower[network] ~ '_mtu') }}
+              vlan_id: {{ lookup('vars', networks_lower[network] ~ '_vlan_id') }}
+              addresses:
+              - ip_netmask:
+                  {{ lookup('vars', networks_lower[network] ~ '_ip') }}/{{ lookup('vars', networks_lower[network] ~ '_cidr') }}
+              routes: {{ lookup('vars', networks_lower[network] ~ '_host_routes') }}
+          {% endfor %}
+          - type: sriov_pf
+            name: eno4
+            numvfs: 10
+            use_dhcp: false
+            promisc: true
```

Comment:
I think this should be similar to [this](https://code.engineering.redhat.com/gerrit/plugins/gitiles/nfv-qe/+/refs/heads/main/tht/17.1/ospd-17.1-geneve-ovn-dpdk-sriov-ctlplane-dataplane-bonding-hybrid-titan119-amd/nic-configs/computeovndpdksriov.yaml.j2) as much as possible. In my opinion we'll need to test this manually first.

Reply:
I can add this in, do you have a pointer to the hardware details specifically the NIC types, number of nics, and potentially how they are networked in NFV deployments?

Reply:
one thing we need to be carful of is while its valid to have ovs-dpdk and sriov on the same host we cannot have the dpdk PF and sriov PF come form the same physical card in general . specifically for some intel nic like x710 where there is one asic for both physical ports haveing one port use the kernel driver and one port using the dpdk userpace driver via vfio-pci is unsupported.

this also applies to some mellanox cards.

so that means we need one physical card for dpdk and a second physical card for legacy sriov in the server where we run this test.